README.rst

```diff
@@ -26,39 +26,53 @@ This is easily achieved by downloading
 Usage Example
 =============
 
-The current `keyboard` module stores key constants which make it easier to
-construct keypress reports for a keyboard device.
+The ``keyboard`` module defines keycode constants and a ``Keyboard`` class to
+construct and send keypress reports for a USB keyboard device.
 
 .. code-block:: python
 
-    import usb_hid
-    import adafruit_hid.keyboard as kbd
-    import time
-
-    report = bytearray(8) # Keyboard reports are always 8 bytes.
-
-    # Devices are initialized earlier so find the one for the keyboard.
-    keyboard = None
-    for device in usb_hid.devices:
-        if device.usage_page is 0x1 and device.usage is 0x06:
-            keyboard = device
-            break
-
-    # The first byte of the report includes a bitfield indicating which
-    # modifiers are pressed. Their bit position is their code's difference from
-    # 0xE0.
-    report[0] |= 1 << (kbd.LEFT_SHIFT - 0xE0)
-    # Normal keys are simply their byte code in bytes 2-7. When fewer than six
-    # keys are pressed then the trailing bytes are zero.
-    report[2] = kbd.A
-    keyboard.send_report(report)
-
-    time.sleep(0.1)
-
-    # Clear the key presses and send another report.
-    report[0] = 0
-    report[2] = 0
-    keyboard.send_report(report)
+    from adafruit_hid.keyboard import Keyboard
+
+    # Find a keyboard device to talk to.
+    kbd = Keyboard()
+
+    # Simulate typing. Press and release the A key (not shifted),
+    # then the B key, then c, then the Enter key.
+    # Type "abc" followed by return.
+    kbd.type("abc\n")
+
+    # Type control-x, then "Abc", then backspace.
+    kbd.type((Keyboard.CONTROL, 'x'), 'Abc', Keyboard.BACKSPACE)
```

Comment:
What do you think about splitting the `type` method out into its own module? It feels to me like a convenience on top of the `press` and `release` mechanics. Splitting it would provide a clear boundary where `press` and `release` can use keycode constants and `type` can use strings.

In general, I like many smaller modules so that people can import just the functionality they want. It also helps with RAM usage during load because each file is smaller individually.

Separately, `type` can then get fancier for those who need it to type something specific. For example, you could actually handle keyboard layouts like `type("Quickly", layout=en_US)` and have it press the right underlying keys.